src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMapping.java

```diff
@@ -55,21 +57,21 @@ public static Integer getHearingDurationAdjournment(
         SscsCaseData caseData,
         ReferenceDataServiceHolder referenceDataServiceHolder
     ) {
-        if (!referenceDataServiceHolder.isAdjournmentFlagEnabled()) {
+        if (!referenceDataServiceHolder.isAdjournmentFlagEnabled()
+            || isNoOrNull(caseData.getAdjournment().getIsAdjournmentInProgress())) {
             return null;
         }
 
-        String nextHearingListingDuration = caseData.getAdjournCaseNextHearingListingDuration();
+        Integer duration = caseData.getAdjournment().getNextHearingListingDuration();
 
-        if (DURATION_TYPE_NON_STANDARD_TIME_SLOT.equals(caseData.getAdjournCaseNextHearingListingDurationType())
-            && isNotBlank(nextHearingListingDuration)) {
-            int duration = Integer.parseInt(nextHearingListingDuration);
+        if (caseData.getAdjournment().getNextHearingListingDurationType() == NON_STANDARD
+            && duration != null) {
 
-            String units = caseData.getAdjournCaseNextHearingListingDurationUnits();
-            if (DURATION_UNITS_SESSIONS.equals(units)
+            AdjournCaseNextHearingDurationUnits units = caseData.getAdjournment().getNextHearingListingDurationUnits();
```

Comment:
Might be good to put this in a separate method (70-77)

Reply:
extracted to handleNonStandardDuration